commands/payment_channel_daemon_test.go

```diff
@@ -205,7 +204,7 @@ func TestPaymentChannelRedeemTooEarlyFails(t *testing.T) {
 		voucher := createVoucherStr(t, d, channelID, types.NewAttoFILFromFIL(111), &payer, uint64(8))
 
 		// Wait for the voucher message to be processed.
-		mustRedeemVoucher(t, targetDaemon, voucher, &target)
+		mustRedeemVoucher(t, targetDaemon, voucher, &target, d)
```

Comment:
`listChannelsAsStrs` should also run from d. Also might be worth renaming `d` and `targetDaemon` to `sender` and `miner` at this point to clarify roles.

Reply:
I feel like payer and payee might be more clear? There isn't any actual storage mining going on here.

Reply:
actually, maybe renaming them to sender and miner and only have one of them mine would improve the situation a lot.

Reply:
Eh, scratch that. It'd definitely mean a lot more changes in the tests.

Reply:
It would be good to establish roles for the the daemons. Possibly just in in this test and the other you changed. The problem is that the transaction is between a payer and a target, but the test requires a role for the sender to send the message and a block miner to mine it. The node sending the message needs to have a key for the sender or target account depending on the message, but otherwise the sender and miner nodes are independent of the other roles.

We started with a single daemon that handled all the roles. We moved to two daemons to resolve some timing issues around sending and mining. For this reason, if we persist with multiple daemons, we should focus on the sender and miner relationship. All messages (payer and target) will be sent from one daemon. All messages will be mined from the other. All messages will be waited on by the sender, and all assertions will be made against sender state.

This is pretty close to what you have here. We should use naming to establish the pattern.